.github/workflows/build_and_test.yml

```diff
@@ -27,7 +27,7 @@ jobs:
             - name: Build
               run: dotnet fake build --parallel 3
             - name: publish build artifacts
-              uses: actions/upload-artifact@v3
+              uses: actions/upload-artifact@v4
               with:
-                  name: fake-artifacts
+                  name: fake-artifacts-${{ matrix.runs-on }}
```

Comment:
The docs at https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md state that artifacts need unique names with the v4 action